### DISCUSSION TITLE
ModuleNotFoundError: No module named 'elftools'

### DISCUSSION THREAD

**Discussion by WajahatRiaz** [zephyrproject-rtos/zephyr#51069] [Q&A] [answered]
**Describe the bug:**
I'm trying to build a simple blinky application for my stm32f103rb board but I'm getting the **ModuleNotFoundError: No module named 'elftools'** error over and over again. I tried it for different boards and QEMU. Still the efforst were in vain. I used the following guide to setup the zephyr project for my Windows based host. 

[https://docs.zephyrproject.org/latest/develop/getting_started/index.html](url)

**To Reproduce:**
`cd %HOMEPATH%\zephyrproject\zephyr`
`west build -p always -c -b nucleo_f103rb samples\basic\blinky`

**Expected behavior:**
A successful build.

**Impact**
Unable to build and run application.

**Logs and console output:**
<img width="656" alt="image" src="https://user-images.githubusercontent.com/61377755/194516743-09cad854-90ae-4ab5-b270-c9628ce8ee36.png">

**Environment (please complete the following information):**
 - OS: Windows
 - Toolchain: Zephyr SDK
 - Version: latest

**Additional context:**
Someone mentioned that it may be some issue with the dependencies so I ran the following commands:
`cd %HOMEPATH%\zephyrproject\zephyr`
`pip3 install --user --upgrade -r scripts/requirements.txt`

All requirements were already installed and updated.

**Comment by stephanosio** [comment]
Please make sure `pyelftools` package is installed.

**Reply by WajahatRiaz** [reply]
> Please make sure `pyelftools` package is installed.

@stephanosio it is installed. As you can see it below:

<img width="667" alt="image" src="https://user-images.githubusercontent.com/61377755/194525287-e9449e38-807e-4281-baa8-444004158462.png">

**Comment by carlescufi** [comment]
You should not use `--user` on Windows, as per the Getting Started Guide.

**Reply by carlescufi** [reply]
Also, try `pip3 list` and see if it shows there

**Comment by WajahatRiaz** [comment, marked as answer]
The main reason behind the issue was that I didn't add the environment variables. Once you add them please restart your machine. I wasted a lot of time over this. Also, I would like to thank @carlescufi for giving me ample time to address my issues.

**Comment by PeterMortensen** [comment]
Re *"...I didn't add the environment variables"*: Which environment variables, specifically? Set to what values? In what context?

[Future readers want to know](https://www.explainxkcd.com/wiki/index.php/979:_Wisdom_of_the_Ancients).

**Comment by PeterMortensen** [comment]
Related: *[No module named 'elftools'](https://github.com/zephyrproject-rtos/zephyr/issues/1392)* (2017)
